### ISSUE TITLE
❓ NextCloud can't use external drive

### ISSUE THREAD

**Issue by LeonardoCT** [alexbelgium/hassio-addons#1446]
name: Help me!
about: the configuration GUI errosout
title: ''
labels: ''
assignees: ''

---

**Describe the solution you'd like**

i can install the nextcloud component with no issue, if i use the default folder /share/nextcloud everything works, but if i try to use my own HHD (of course, i don't want do fill up my HA installation drive) it says it can't access the folder i've setup the external HDD via the "network archives" in HA settings:
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/be3329b8-5fe8-4393-987a-9a711af3a90d)
And the drive happears under:
/share/Specchio/
But if i try to use it under nextcloud:
/share/Specchio/nextcloud

![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/6fe8745e-f1e8-4d2f-bc4b-c69c3ad5a603)

Tranlation:
Error:
Unable to create or write to data folder /share/Mirror/nextcloud/

Is there any other configuration that i'm missing?

**Comment by alexbelgium** [comment]
Hi, nextcloud runs with permissions 1000:1000 so you must first create a folder a chown it to 1000:1000 before being able to use it

**Comment by LeonardoCT** [comment]
Hi, thank you for the fast response, i've created the folder following your specifications:

mkdir nextcloud && chown 1000:1000 nextcloud

but still, the same error:
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/72eb30cb-1b30-4811-8460-7b2abe715cdf)

**Comment by alexbelgium** [comment]
Mmh perhaps you could try using my own system to mount drive otherwise :-) it will then be mounted directly from the nextcloud addons : https://github.com/alexbelgium/hassio-addons/wiki/Add-ons-feature-:-mounting-drives

**Comment by LeonardoCT** [comment]
i'm already using the "Samba NAS" addon will this method interfere with it?

**Comment by LeonardoCT** [comment]
i see that your method supports both hardware "localdisks" and "software" via SMB, i will try SMB first, since it looks like the best option, but i think it will decrese performance, right? like with the SMB hoverhead?

**Comment by alexbelgium** [comment]
> i'm already using the "Samba NAS" addon will this method interfere with it?

It's initially the same code that I adapted over the years, so perfectly parallel ! If you have a physical drive connected I would highly recommend that you use it instead of SMB, indeed it would adde unecessary steps although I don't know if it would affect much

**Comment by LeonardoCT** [comment]
Drive is found:
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/c0f35689-ffdb-4cb9-944c-3c5debf8c470)

Mount is successful:
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/385cbfba-ece1-471f-8ed1-0920390d2b72)

STILL the same error:
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/ea40961e-f7de-41dd-8cde-439b76ca7fb7)

i even did the 1000:1000 permission

Did i get the path wrong?

**Comment by alexbelgium** [comment]
Mmh I'm really at a loss... and there is no specific messages in the addon logs ? Perhaps you could try with sqlite to check if the issue is with the disk mount or mariadb ?

**Comment by LeonardoCT** [comment]
Hi, i did try with the default sqlite, but still no luck, got the same error, beside... if use the default folder, both mariadb and sqlite work, i've attached a full copy of the log, but i didn't see anything... all that i can see is that the drive is mounted succesfully
`[file.txt](https://github.com/user-attachments/files/15957835/file.txt)`

**Comment by alexbelgium** [comment]
Thanks, and for the full log. I see that there is no error anywhere... Looking in Google for your translated error there is many things that pop-up but we can already eliminate the database type (as error common for sqlite and mariadb) and the add-on itself if it worked using a local drive... Did you do a full install/uninstall between both test?

I wonder if the permission of the parent directory matters. I mean you could try to do a nextcloud/nextcloud folder on your sda1 drive, got 777 permissions to both and try again... I don't see much other things to try for the moment

**Comment by LeonardoCT** [comment]
still no luck...
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/f366355c-58b0-418c-a509-a2e321804a96)

i also did a full install and uninstall of both nextcloud and mariadb, i will now try to use SMB as a way to mount the drive

**Comment by alexbelgium** [comment]
That's really crazy especially as you have tried both mine and HA's mounting system...

**Comment by LeonardoCT** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/a2918606-c5ce-4528-b897-5efb603417c7)

IT DID WORK!

but a strange issue that may be get us on the right track...

![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/088e8ddd-6776-4f20-b9d2-0bdb95d77fa1)

And then... it let's me log in and use the interface:

![image](https://github.com/alexbelgium/hassio-addons/assets/20495117/556f8406-c1a7-4a94-89ea-bf2e7fb32f4d)

and files are stored on the HDD

This entire time i've been using the web terminal addon for issuing the commands, what we let the container create the folder by it self with all the permission issued by the container?

**Comment by LeonardoCT** [comment]
But i guess... now i have the addon running, would like to remove the extra SMB jumps as the interface feels a little sluggish

**Comment by alexbelgium** [comment]
Amazing!!! The issue is that the only way to switch would be to have your localdrive mounted to /mnt/nextcloud... Otherwise it would require manual operations to rename the data folder in config, and in the database 

Indeed the add-on creates all structure!

**Comment by LeonardoCT** [comment]
> Amazing!!! The issue is that the only way to switch would be to have your localdrive mounted to /mnt/nextcloud... Otherwise it would require manual operations to rename the data folder in config, and in the database
> 
> Indeed the add-on creates all structure!

i think i didn't understand what you where trying to say here, can you rephrase it please?

**Comment by LeonardoCT** [comment]
I acctualy found a new issue, EVERY addon can't installed with the following error:
"Error: This app cannot be enabled because it makes the server unstable"
But reddit has allready worked out the issue:
https://www.reddit.com/r/NextCloud/comments/1bqf8wp/2804_error_this_app_cannot_be_enabled_because_it/

Maybe the container needs updating ?

**Comment by alexbelgium** [comment]
Hehe I meant that the data location is recorded both in the config file of nextcloud, and the database. If you find a way to have the exact same path with smb and localdisk it will work. Otherwise it would need manual modification which is quite complex.

For the app not installing, alas the solution was actually proposed by myself on the linuxserver container and already enabled in the add-on : https://github.com/linuxserver/docker-nextcloud/issues/425#issuecomment-208810395... Do you have several layers of proxies? Nextcloud has implemented new security criteria that are a mess for users... However activating apps from command line with occ works

**Comment by alexbelgium** [comment]
I'll push the fix now

**Comment by LeonardoCT** [comment]
> If you find a way to have the exact same path with smb and localdisk it will work. Otherwise it would need manual modification which is quite complex.
>
now i understand, thank you for the push,

as for closing the issue the "solution" was to use SMB since soon i will get a rack mounted server where i will have a different aproach for data storage i won't look too much into it as this was a way to demo nextcloud for me

**Comment by alexbelgium** [comment]
Yes, I have a script injector. If you type `mnt /dev/sda1 /mnt/nextcloud` for example you can mount you drive to exactly where your want. The location of the script inscrire is described in the log at add-on start. Here is a very fast wiki : https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

**Comment by yribollen** [comment]
For whom it may concern.... To solve this insert folowing line in /addon_configs/?????????_nextcloud_ocr/nextcloud-ocr.sh


sudo chown 1000:1000 /mnt/YOUR_EXTERNAL_DRIVE/PATH/TO/DIRECTORY

**Comment by alexbelgium** [comment]
Thanks ! In theory it should be done automatically https://github.com/alexbelgium/hassio-addons/blob/02958e369b6e9e7cbd196bf24a3af62d5acfc299/nextcloud/rootfs/etc/cont-init.d/01-folders.sh#L58 isn’t it for you ?
